### ISSUE TITLE
Add  a BRANCH parameter when importing a git module in config/_default/hugo.toml

### ISSUE THREAD

**Issue by vinceyoumans** [gohugoio/hugo#11928]
Use of MODULES hosted on Git repos is an awesome feature.  For example...

```text
[[module.imports]]
    # path = 'github.com/bingo/tl01-bigme-content@v02'
    path = 'github.com/bingo/tl01-bigme.com-content'
    # branch = 'v02'.   BRANCH will always be MAIN.  not an option to pick branch in hugo config
    # branch = 'main'
    [[module.imports.mounts]]
      # source = "v02/content/en"
      source = "v03/content"
      target = "content"
```

In this code snip, content is maintained separately but still mounted.

# PROBLEM
Only the MAIN branch of the above CONTENT repo in imported.  The 
    # branch = 'main'.  
does nothing.  But It would be very cool if we could include BRANCH into the Path.

I would be very happy to do this feature if I can get some feed back on what would need to be done or if I am missing something.

cheers

**Comment by jmooring** [comment]
After the initial import, you can:

```
hugo mod get github.com/foo/bar@somebranch
```

But it would be convenient to handle that in config.

**Comment by bep** [comment]
I agree that it would be useful, but I'm not sure how simple it is to implement. The operations we do on regular Hugo builds are, I think:

* `go mod download`
* `go list -m -json`

Both of the above operates on the content of your go.mod/work file(s).

**Comment by jmooring** [comment]
This was addressed in v0.150.0:
<https://gohugo.io/configuration/module/#version>
